plugins/inputs/fritzbox/fritzbox.go

```diff
@@ -0,0 +1,454 @@
+	igdServices, err := deviceClient.ServicesByType(tr064.IgdServiceSpec, "WANCommonInterfaceConfig")
+	if err != nil {
+		return err
+	}
+	if len(igdServices) == 0 {
+		return fmt.Errorf("")
+	}
+	igdServiceClient := &igdicfg.ServiceClient{
+		TR064Client: deviceClient,
+		Service:     igdServices[0],
+	}
+	addonInfos := &igdicfg.GetAddonInfosResponse{}
+	err = igdServiceClient.GetAddonInfos(addonInfos)
+	if err != nil {
+		return err
+	}
+	strconv.ParseUint(addonInfos.NewX_AVM_DE_TotalBytesSent64, 10, 64)
+	totalBytesSent, err := strconv.ParseUint(addonInfos.NewX_AVM_DE_TotalBytesSent64, 10, 64)
+	if err != nil {
+		return err
+	}
+	totalBytesReceived, err := strconv.ParseUint(addonInfos.NewX_AVM_DE_TotalBytesReceived64, 10, 64)
+	if err != nil {
+		return err
+	}
+	tags := make(map[string]string)
+	tags["source"] = serviceClient.TR064Client.DeviceUrl.Hostname()
+	tags["service"] = serviceClient.Service.ShortId()
+	fields := make(map[string]interface{})
+	fields["layer1_upstream_max_bit_rate"] = commonLinkProperties.NewLayer1UpstreamMaxBitRate
+	fields["layer1_downstream_max_bit_rate"] = commonLinkProperties.NewLayer1DownstreamMaxBitRate
+	fields["upstream_current_max_speed"] = commonLinkProperties.NewX_AVM_DE_UpstreamCurrentMaxSpeed
+	fields["downstream_current_max_speed"] = commonLinkProperties.NewX_AVM_DE_DownstreamCurrentMaxSpeed
+	fields["total_bytes_sent"] = totalBytesSent
+	fields["total_bytes_received"] = totalBytesReceived
+	acc.AddFields("fritzbox_wan", fields, tags)
+	if plugin.Log.Level().Includes(telegraf.Debug) {
+		plugin.logMetric("fritzbox_wan", tags, fields)
+	}
+	return nil
+}
+
+func (plugin *Fritzbox) gatherPppInfo(acc telegraf.Accumulator, deviceClient *tr064.Client, service tr064.ServiceDescriptor) error {
+	serviceClient := wanpppconn.ServiceClient{
+		TR064Client: deviceClient,
+		Service:     service,
+	}
+	info := &wanpppconn.GetInfoResponse{}
+	err := serviceClient.GetInfo(info)
+	if err != nil {
+		return err
+	}
+	tags := make(map[string]string)
+	tags["source"] = serviceClient.TR064Client.DeviceUrl.Hostname()
+	tags["service"] = serviceClient.Service.ShortId()
+	fields := make(map[string]interface{})
+	fields["uptime"] = info.NewUptime
+	fields["upstream_max_bit_rate"] = info.NewUpstreamMaxBitRate
+	fields["downstream_max_bit_rate"] = info.NewDownstreamMaxBitRate
+	acc.AddFields("fritzbox_ppp", fields, tags)
+	if plugin.Log.Level().Includes(telegraf.Debug) {
+		plugin.logMetric("fritzbox_ppp", tags, fields)
+	}
+	return nil
+}
+
+func (plugin *Fritzbox) gatherDslInfo(acc telegraf.Accumulator, deviceClient *tr064.Client, service tr064.ServiceDescriptor) error {
+	serviceClient := wandslifconfig.ServiceClient{
+		TR064Client: deviceClient,
+		Service:     service,
+	}
+	info := &wandslifconfig.GetInfoResponse{}
+	err := serviceClient.GetInfo(info)
+	if err != nil {
+		return err
+	}
+	if info.NewStatus != "Up" {
+		return nil
+	}
```

Comment:
Maybe people still would like to get service name if it is down? What are the values in that case? All zeros? (`status` could be a tag so it can be filtered on it afterwards..)

(and will there always be only 1 DSL port?)

Reply:
Good suggestion. Made status a tag. Stats will be zero, as it makes no sense to query them.
So far all device have a maximum of 1 DSL port. If there would be 2 port model in the future, a 2nd service (with a separate service id) will be recorded. So multiple ports can be distinguished.